xdsl/dialects/linalg.py

```diff
@@ -201,6 +168,52 @@ def get_static_loop_ranges(self) -> tuple[int, ...]:
         static_shapes = self.get_static_shapes()
         return shapes_to_loops.eval(static_shapes, [])
 
+
+@irdl_op_definition
+class GenericOp(LinalgOperation):
+    name = "linalg.generic"
+
+    inputs = var_operand_def()
+    outputs = var_operand_def(base(ShapedType))
+
+    res = var_result_def(AnyTensorType)
+
+    body = region_def("single_block")
+
+    # Trait attributes
+    indexing_maps = prop_def(ArrayAttr[AffineMapAttr])
+    iterator_types = prop_def(ArrayAttr[IteratorTypeAttr])
+    doc = opt_prop_def(StringAttr)
+    library_call = opt_prop_def(StringAttr)
+
+    irdl_options = [AttrSizedOperandSegments(as_property=True)]
+
+    def __init__(
+        self,
+        inputs: Sequence[SSAValue],
+        outputs: Sequence[SSAValue],
+        body: Region,
+        indexing_maps: Sequence[AffineMapAttr] | ArrayAttr[AffineMapAttr],
+        iterator_types: Sequence[Attribute] | ArrayAttr[Attribute],
+        result_types: Sequence[Attribute] = (),
+        doc: StringAttr | None = None,
+        library_call: StringAttr | None = None,
+    ) -> None:
+        super().__init__(
+            operands=[inputs, outputs],
+            result_types=[result_types],
+            properties={
+                "indexing_maps": ArrayAttr(indexing_maps),
```

Comment:
```suggestion
                "indexing_maps": ArrayAttr.get(indexing_maps),
```
I know its not the content of this change but just realised that this works as `ArrayAttr` is a `Sequence` so it rebuilds itself.

Reply:
I realised this as an issue when defining `Data.get`. Perhaps `ArrayAttr` should have its own `get` anyway.